my-vault/internal/handlers/handlers.go

```diff
@@ -0,0 +1,181 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/bmasaiti/go-projects/my-vault/internal/domain"
+	"github.com/google/uuid"
+)
+
+//var db = storage.NewInMemorySecretRepo()
+//return domain.Secret{}, fmt.Errorf("secret with ID '%s' not found", secretId)
+//fmt.Errorf("no secrets found in the secrets store")
+var ErrNotFound = errors.New("not found")
+type SecretsRepository interface {
+	PutNewSecret(secret domain.Secret) error
+	GetScretsById(Id string) (domain.Secret, error)
+	DeleteSecretByID(Id string) (string, error)
+	ListAllSecrets() ([]domain.Secret, error)
+}
+
+// Couldn't mock the tests , need to inject the db
+type SecretHandler struct {
+	DB SecretsRepository
+}
+
+type CreateSecretRequest struct {
+	Name  string            `json:"name"`
+	KVMap map[string]string `json:"kv_map"`
+}
+
+type CreateSecretResponse struct {
+	Message string `json:"message"`
+	Id      string `json:"id"`
+	Name    string `json:"name"`
+}
+type GetSecretResponse struct {
+	Message string            `json:"message"`
+	Id      string            `json:"id"`
+	Name    string            `json:"name"`
+	KVMap   map[string]string `json:"kv_map"`
+}
+
+type ListSecretsResponse struct {
+	Message string          `json:"message"`
+	Secrets []domain.Secret `json:"Secrets"`
+}
+
+func GenerateUUID() string {
+	newUUID, err := uuid.NewV7()
+	if err != nil {
+		fmt.Println("Error generating UUID:", err)
+		return err.Error()
+	}
+	return newUUID.String()
+}
```

Comment:
This is just returning the error as a string, which makes it very difficult for the caller to know when something went wrong. Instead, return a separate error variable:
```suggestion
func GenerateUUID() (string, error) {
	newUUID, err := uuid.NewV7()
	if err != nil {
		return "", err
	}
	return newUUID.String(), nil
}
```